### PR TITLE
DOC-5889 - overview of sync gateway

### PR DESCRIPTION
**Pull request by jamesnocentini**
https://issues.couchbase.com/browse/DOC-5889

Changes:

- Add diagram
- Add text

### CODE REVIEW

File: modules/ROOT/assets/images/cbm-architecture.svg [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+            </g>
+            <g id="intranet" transform="translate(756.000000, 96.000000)">
+                <path d="M10,0 L1783,0 C1788.52285,-1.01453063e-15 1793,4.4771525 1793,10 L1793,1143 C1793,1148.52285 1788.52285,1153 1783,1153 L10,1153 C4.4771525,1153 6.76353751e-16,1148.52285 0,1143 L0,10 C-6.76353751e-16,4.4771525 4.4771525,1.01453063e-15 10,0 Z" id="Rectangle" stroke="#000000" stroke-width="4" stroke-dasharray="15"></path>
+                <g id="sdks" transform="translate(511.000000, 45.000000)">
+                    <rect id="Rectangle" stroke="#000000" stroke-width="3" fill="#FFFFFF" x="0" y="0" width="359" height="311" rx="10"></rect>
+                    <text id="Web-Server" font-family="Helvetica" font-size="30" font-weight="normal" letter-spacing="-0.18461538" fill="#000000">
+                        <tspan x="100.997296" y="48">Web Server</tspan>
+                    </text>
+                    <text id="80" font-family="Helvetica" font-size="30" font-weight="normal" letter-spacing="-0.18461538" fill="#000000">
+                        <tspan x="162.500045" y="83">80</tspan>
```
Comment (simon-dew) [inline modules/ROOT/assets/images/cbm-architecture.svg]: Will all web clients communicate with their web server using port 80? Do we need to add "(typical)"?

File: modules/ROOT/pages/introduction.adoc [hunk 2]
```diff
@@ -8,6 +8,10 @@ Couchbase Mobile brings the power of NoSQL to the edge and is comprised of three
 * xref:sync-gateway::introduction.adoc[Sync Gateway], an internet-facing synchronization mechanism that securely syncs data between mobile clients and server, and
 * xref:server:introduction:intro.adoc[Couchbase Server], a highly scalable, distributed NoSQL database platform
 
+The following diagram describes the architecture for a typical deployment comprised of Couchbase Lite, Couchbase Server SDKs, Sync Gateway and Couchbase Server.
```
Comment (simon-dew) [inline modules/ROOT/pages/introduction.adoc]: Oxford comma as per style guide

> Sync Gateway, and Couchbase Server

**Review by simon-dew — APPROVED** [review]
👍 LGTM, just 2 minor comments

File: modules/ROOT/assets/images/cbm-architecture.svg (hunk 1 above)
Comment (jamesnocentini) [reply modules/ROOT/assets/images/cbm-architecture.svg]: - I've added the "port" suffix to make clear we're talking about the port numbers.
- Added port 443 for https

ports 80/443 are the defaults for http/https respectively, similarly to how 8091 is the default for CBServer and 4984 for Sync Gateway. I'm going to create a new PR to say the diagram is for information only and different architectures could be designed.

https://github.com/couchbase/docs-sync-gateway/commit/9679302d8ae248fc438940342e02910a312cb907?short_path=6e22201#diff-6e222012ced9fbb0cafa593feb59b5c8

File: modules/ROOT/pages/introduction.adoc (hunk 2 above)
Comment (jamesnocentini) [reply modules/ROOT/pages/introduction.adoc]: https://github.com/couchbase/docs-sync-gateway/pull/113/commits/81d0d26d7950e9e3d7b8d4c4c2c949039e6e48b4

**Comment by jamesnocentini** [comment]
Will merge now. Further enhancements can be applied in the future.

**Comment by jamesnocentini** [comment]
2.5 https://github.com/couchbase/docs-sync-gateway/commit/b6d7f1bdf1b9b7b896fc2c4dc7c2cd169692c51f
2.1 https://github.com/couchbase/docs-sync-gateway/commit/6c32246b7fd3b6c9a187981806c465ac6f261b45

**Pull request merged**